run/opencl/opencl_misc.h

```diff
@@ -38,6 +38,15 @@ typedef uint32_t host_size_t;
 #define NULL ((void*)0)
 #endif
 
+/* Change the below to '#if 0' if your platform can't handle __builtin_expect */
+#if 1
```

Comment:
It's not great to require code edit on non-LLVM. Maybe there's an `#ifdef` we can check to detect LLVM-based OpenCL?

Reply:
I'll investigate. Perhaps even the likely/unlikely macros already exist where applicable (alas I doubt that), so we could just check for them. Also, I'm not sure that thing is very important with current code, I should test that as well.